sdk/core/azure-core/test/ut/curl_session_test.cpp

```diff
@@ -41,4 +42,40 @@ namespace Azure { namespace Core { namespace Test {
 
     EXPECT_NO_THROW(session->Perform(Azure::Core::GetApplicationContext()));
   }
+
+  TEST_F(CurlSession, chunkResponseSizeZero)
+  {
+    // chunked response with no content and no size
+    std::string response("HTTP/1.1 200 Ok\r\ntransfer-encoding: chunked\r\n\r\n\n\r\n");
```

Comment:
Add more test permutations to cover all the scenarios you listed in the issue:

> Server can respond either with `0\r\n` or `\n\r\n` or `.*\r\n` (where .* are non number digitis) and these responses means basically the same, `a chunk with 0 size`.

Reply:
one case is enough to cover all, because we rely on std::stoul

Reply:
That's an implementation detail though.